From: Kalle Sommer Nielsen <kalle@php.net>
Date: Wed Aug 3 17:33:18 2016 +0200
Subject: [PATCH] Fixed bug #72682 (exif_read_data() fails to read all data for some images)

This is fixed by adding DJI signatures to the MAKERNOTE and its supported tags, list is credits to ExifTool documentation.

5.5 backport: https://github.com/shivammathur/php-src-backports/commit/c5c54656b46b94e0e38ad2b3c7101b792927405e
---
diff --git a/ext/exif/exif.c b/ext/exif/exif.c
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -971,6 +971,20 @@ static tag_info_array tag_table_VND_OLYMPUS = {
   TAG_TABLE_END
 };
 
+static tag_info_array tag_table_VND_DJI = {
+  { 0x0001, "Make"}, 
+  { 0x0003, "SpeedX"}, 
+  { 0x0004, "SpeedY"}, 
+  { 0x0005, "SpeedZ"}, 
+  { 0x0006, "Pitch"}, 
+  { 0x0007, "Yaw"}, 
+  { 0x0008, "Roll"}, 
+  { 0x0009, "CameraPitch"}, 
+  { 0x000a, "CameraYaw"}, 
+  { 0x000b, "CameraRoll"}, 
+  TAG_TABLE_END
+};
+
 typedef enum mn_byte_order_t {
 	MN_ORDER_INTEL    = 0,
 	MN_ORDER_MOTOROLA = 1,
@@ -1002,6 +1016,7 @@ static const maker_note_type maker_note_array[] = {
   { tag_table_VND_NIKON,     "NIKON",                   NULL,  "Nikon\x00\x01\x00",        8,  8,  MN_ORDER_NORMAL,   MN_OFFSET_NORMAL},
   { tag_table_VND_NIKON_990, "NIKON",                   NULL,  NULL,                       0,  0,  MN_ORDER_NORMAL,   MN_OFFSET_NORMAL},
   { tag_table_VND_OLYMPUS,   "OLYMPUS OPTICAL CO.,LTD", NULL,  "OLYMP\x00\x01\x00",        8,  8,  MN_ORDER_NORMAL,   MN_OFFSET_NORMAL},
+  { tag_table_VND_DJI,       "DJI",                     NULL, NULL,                        0,  0,  MN_ORDER_NORMAL,  MN_OFFSET_NORMAL}
 };
 /* }}} */
 
